### PR TITLE
*: implement scheduler for adding index

### LINKED ISSUE #42754: Implement scheduler for the partition table.

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #42754 

Problem Summary:
This PR implements a scheduler for adding index.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/scheduler.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	logutil.BgLogger().Info("[ddl] lightning init subtask exec env")
+	d, _ := GetDDL().(*ddl)
+
+	bc, err := ingest.LitBackCtxMgr.Register(d.ctx, b.index.Unique, b.job.ID, b.job.ReorgMeta.SQLMode)
+	if err != nil {
+		logutil.BgLogger().Warn("[ddl] lightning register error", zap.Error(err))
+		return err
+	}
+	b.bc = bc
+	d.newReorgCtx(b.job.ID, nil, &meta.Element{ID: b.index.ID, TypeKey: b.eleTypeKey}, 0)
```
Comment (lcwangchao) [inline ddl/scheduler.go:116]: Should use `genBackfillJobReorgCtxID(b.job.ID)` as ReorgCtx id

**Review by lcwangchao — APPROVED** [review]

File: ddl/scheduler.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+	err = ingestScheduler.setupWorkers()
+	if err != nil {
+		logutil.BgLogger().Error("[ddl] setup workers error", zap.Error(err))
+		return nil, err
+	}
+
+	for {
+		kvRanges, err := splitTableRanges(b.ptbl, d.store, startKey, endKey, backfillTaskChanSize)
+		if err != nil {
+			return nil, err
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply ddl/scheduler.go:161]: No

**Pull request merged**